lib/auth.js

```diff
@@ -0,0 +1,453 @@
+'use strict';
+const _ = require('lodash'),
+  passport = require('passport'),
+  users = require('./services/users'),
+  db = require('./services/db'),
+  references = require('./services/references'),
+  session = require('express-session'),
+  flash = require('express-flash'),
+  handlebars = require('handlebars'),
+  basicAuth = require('basic-auth'),
+  fs = require('fs'),
+  path = require('path');
+
+/**
+ * determine if a route is protected
+ * protected routes are ?edit=true and any method other than GET
+ * @param {object} req
+ * @returns {boolean}
+ */
+function isProtectedRoute(req) {
+  return !!req.query.edit || req.method !== 'GET';
```

Comment:
What is concerning is that we are saying any site built on amphora cannot have an `edit` query param. Would be nice to leave the query params open.

I am ok with this for this PR, but we might want to consider something like a `locals.isProtected` property to indicate that login is required. The sites repo could add `isProtected` to any request that had the `edit` queryparam. That way the individual implementation of clay controls which queryparams are protected.